integration-tests/common/src/test/kotlin/me/tatarka/inject/test/AbstractComponentTest.kt

```diff
@@ -1,3 +1,5 @@
+@file:Suppress("UnnecessaryAbstractClass")
```

Comment:
Might be worth having separate main and test configs, have the test code be more lenient since it seems to incorrectly be flagging test cases.

Reply:
Upon closer look, there were only 2 checks that the tests were triggering, so I went with a lighter-weight approach of excluding the test files in a few checks. The extra `@Suppress` sites are now gone. Tell me if this works.